### PR TITLE
added ENS resolver hooks, display results in the console   (Ready for review)

### PR DESCRIPTION
**Pull request by Seroxdesign**
## Progress

- [x]  Added two functions in the hooks folder. one resolves ENS to address and one resolves address to ENS

### Remaining work

What's left:

- [x]  Adding checks for search terms to ensure we call the right method
- [x]  Adding a debounce to minimise spam to RPC
- [x]  Showing results under search bar and on the account view page

### CODE REVIEW

**Comment by Seroxdesign** [comment]
# THIS PR IS A WORK IN PROGRESS

**Comment by kasparkallas** [comment]
Glad to see interest in contributing to the Console!

Some comments on how to solve this story:

* We use [RTK-Query](https://redux-toolkit.js.org/rtk-query/overview) to handle cache'ing and some request debounce'ing. Ethers' `JsonRpcBatchProvider` & `FallbackProvider` are used for further throttling & robustness.
* We implemented a solution for the upcoming Dashboard and would like to re-use it in the Console. If the solution is not good then we'd like to know why. I put the snippets of the code in another ENS pull request: https://github.com/superfluid-finance/superfluid-console/pull/88#issuecomment-1159093336 Effectively this bounty can be grabbed by just following these snippets.

**Comment by Seroxdesign** [comment]
Still reading through all the snippets, and deciding how to fit them in. Expecting results by July 9th.

**Comment by Seroxdesign** [comment]
@kasparkallas @vmichalik @sunnyjaycer 

Ready for review

File: src/components/AddressBook.tsx [hunk 1]
```diff
@@ -23,18 +24,24 @@ import {
   getEntryId,
 } from "../redux/slices/addressBook.slice";
 import { Network } from "../redux/networks";
+import { ensApi } from "../redux/slices/ensResolver.slice";
 import { ethers } from "ethers";
 
 export const AddressBookButton: FC<{
   network: Network;
   address: string;
+  description: any;
```
Comment (kasparkallas) [inline src/components/AddressBook.tsx:33]: What's a "description". Why type "any"?

File: src/redux/slices/addressBook.slice.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import {
 import { Network } from '../networks';
 import {RootState} from "../store";
 import {REHYDRATE} from "redux-persist";
+import { add } from 'lodash';
```
Comment (kasparkallas) [inline src/redux/slices/addressBook.slice.ts:8]: Not used?

File: src/redux/store.ts [hunk 3]
```diff
@@ -61,7 +62,9 @@ export const makeStore = wrapMakeStore(() => {
       [rpcApi.reducerPath]: rpcApi.reducer,
       [sfSubgraph.reducerPath]: sfSubgraph.reducer,
       [themePreferenceSlice.name]: themePreferenceSlice.reducer,
-      [addressBookSlice.name]: addressBookReducer
+      [addressBookSlice.name]: addressBookReducer,
+      [addressBookSlice.name]: addressBookReducer,
```
Comment (kasparkallas) [inline src/redux/store.ts:66]: Double reducer

File: package.json [hunk 4]
```diff
@@ -33,6 +33,7 @@
     "next-redux-cookie-wrapper": "^2.1.2",
     "next-redux-wrapper": "^7.0.5",
     "react": "^18.0.0",
+    "react-blockies": "^1.4.1",
```
Comment (kasparkallas) [inline package.json:36]: Didn't see blockies anywhere

**Review by kasparkallas — COMMENTED** [review]
Did a quick tour and seems to work overall! :+1:

Couple of comments:
* doesn't show spinner when still fetching ENS
* when on an account page and then going to another account's page then previous avatar is shownfor a short period of time (use RTK-query's `.currentData` over `.data` or set a _key_ for the page [use address as key for example])
* when both ENS and address book name are specified then the behaviour is weird with both showing and the address not showing
* UI is a bit messy and doesn't follow https://mui.com/material-ui/

File: src/pages/[_network]/accounts/[_id].tsx [hunk 5]
```diff
@@ -64,12 +65,15 @@ import ellipsisAddress from "../../../utils/ellipsisAddress";
...
   const address = useContext(IdContext);
-
   const accountQuery = sfSubgraph.useAccountQuery({
     chainId: network.chainId,
     id: address,
   });
 
+  const ensAddressQuery = ensApi.useLookupAddressQuery(address);
+
+  const ensName = ensAddressQuery.data?.name
```
[the remark this answers is not in the document]
Comment (Seroxdesign) [reply src/pages/[_network]/accounts/[_id].tsx]: Done

File: package.json (hunk 4 above)
Comment (Seroxdesign) [reply package.json]: Removed unused package and types

File: src/redux/slices/addressBook.slice.ts (hunk 2 above)
Comment (Seroxdesign) [reply src/redux/slices/addressBook.slice.ts]: Removed

**Comment by Seroxdesign** [comment]
I believe most problems have been resolved, although I might need to figure out how to display addressBook name and ENS name

**Comment by Seroxdesign** [comment]
> Did a quick tour and seems to work overall! 👍
> 
> Couple of comments:
> 
>     * doesn't show spinner when still fetching ENS
> 
>     * when on an account page and then going to another account's page then previous avatar is shownfor a short period of time (use RTK-query's `.currentData` over `.data` or set a _key_ for the page [use address as key for example])
> 
>     * when both ENS and address book name are specified then the behavior is weird with both showing and the address not showing
> 
>     * UI is a bit messy and doesn't follow https://mui.com/material-ui/

- Switched to using currentData

- Added conditionals to avoid rendering both address book name and ens name at the same time

- Slightly improved UI on account page

- Spinner isn't loading immediately because it takes a second to resolve ENS name to address for search

@kasparkallas Let me know if anything else needs work

**Comment by Seroxdesign** [comment]
So seems like there's an issue I am not handling right, I don't know what the best way to check name before I resolve to address

**Review by vmichalik — APPROVED** [review]
looks good to me!

**Comment by elvijsTDL** [comment]
Doesn't work on Brave and Firefox browsers
https://user-images.githubusercontent.com/77115130/179575974-d69c8ae3-4162-4780-9939-70d311077a74.mov

The non-ens address in at the top of the page is alligned a little bit off now and it is smaller, here it is now
Screenshot 2022-07-18 at 21 20 08

And here is the before
Screenshot 2022-07-18 at 21 21 37


@vmichalik Should we also include showing the ENS names in the tables and using them in filters in this ticket?

**Review by elvijsTDL — CHANGES_REQUESTED** [review]
Was on a slower network and the loading spinner doesn't show up when the ENS lookup is happening, actually thought the ENS lookup didn't work on Brave and Firefox because of that, could we please have one while the lookup is still happening

**Comment by Seroxdesign** [comment]
> The non-ens address in at the top of the page is alligned a little bit off now and it is smaller, here it is now Screenshot 2022-07-18 at 21 20 08
> 
> And here is the before Screenshot 2022-07-18 at 21 21 37
> 
> @vmichalik Should we also include showing the ENS names in the tables and using them in filters in this ticket?

I changed the size of the text to comply with the ENS frontend standard, 

image

An example of how this is relevant, I wanted the ENS to be bigger than the address, although this can be changed back to the original style

**Comment by Seroxdesign** [comment]
> Was on a slower network and the loading spinner doesn't show up when the ENS lookup is happening, actually thought the ENS lookup didn't work on Brave and Firefox because of that, could we please have one while the lookup is still happening , here is a video:
> Screen.Recording.2022-07-18.at.21.40.42.mov
> 
> Also address book entries now show the input string too instead of just the result: Screenshot 2022-07-18 at 22 00 44

The text is a simple fix, I will remove showing the text on top of the address and instead only showed related addresses on ENS search.

As for the spinner, I tried fixing this yesterday but I am still trying to figure out the best way to do this without changing how the spinner renders. I will try again today and if I can't find a solution I will request support

**Comment by Seroxdesign** [comment]
Before:

image
image


After:

image
image

What's left:

- []  Figuring out how to show spinner correctly.

**Comment by Seroxdesign** [comment]
@elvijsTDL @kasparkallas I am having trouble with the spinner task, I would appreicate any pointer or advice,

**Comment by Seroxdesign** [comment]
> Was on a slower network and the loading spinner doesn't show up when the ENS lookup is happening, actually thought the ENS lookup didn't work on Brave and Firefox because of that, could we please have one while the lookup is still happening , here is a video:
> Screen.Recording.2022-07-18.at.21.40.42.mov
> 
> Also address book entries now show the input string too instead of just the result: Screenshot 2022-07-18 at 22 00 44

image

Fixed using isFetching on ensQuery

**Comment by vmichalik** [comment]
@Seroxdesign, we really appreciate your efforts- you took to the task quickly, were responsive to feedback and suggestions and kept making tweaks to improve, so I'm more than happy to see this bounty paid out. 

With Dashboard V2 still under such active development we likely won't have time to do a Console release containing this feature for a few weeks. However, we hope you enjoy the bounty payout, learnt more about ENS and the SDK-Redux and stick around for future bounties too🥇 

Will DM you on Discord for next steps

**Comment by elvijsTDL** [comment]
LGTM  👍

**Review by elvijsTDL — APPROVED** [review]

**Review by kasparkallas — APPROVED** [review]
When there's an address book entry then ENS name is not displayed anywhere. Not sure if I like it... The ENS name should still be visible somewhere but let's handle that in the next iteration.

**Pull request merged**